### ISSUE TITLE
Image inside svg

### ISSUE THREAD

**Issue by borgues** [QuestPDF/QuestPDF#903]
Eu tenho uma imagem em svg que contém uma imagem em base 64 como background. Essa imagem será inserido no pdf ocupando todo a página. O svg é carregado, mas a imagem de fundo que esta dentro do svg não é carregado, dado a seguinte mensagem: can't render image: load image failed

Thats is the svg file:
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xml="http://www.w3.org/XML/1998/namespace" width="420" height="280" viewBox="0, 0, 420, 280">
  <g>
    <image width="420" height="280" xlink:href="data:image/jpeg;base64, <image data> " />
    <text x="209.0788" y="135.9987" text-anchor="middle" font-family="Corbel" font-size="15">Teste</text>
  </g>
</svg>

The expected result should be a PDF file with the image, as shown below:

![image](https://github.com/QuestPDF/QuestPDF/assets/12438437/92261a88-2860-48cb-b4e0-c59d7cc28464)

**Environment**
I'm using the latest version of the library on Windows 11 x64

Thank´s

**Comment by MarcinZiabek** [comment]
Thank you for sharing this issue 😄

I have investigated the bug, and it should be fixed in the `2024.3.10` release. Could you please give it a try?

**Comment by lukasvecerka23** [comment]
@MarcinZiabek Hello, we're facing similar issue. We have SVG which contains png images. When I use the image in PDF I can see everything from the SVG except the images. Do you know what could be problem?

.NET version: .NET Framework 4.7.2
QuestPDF version: 2024.6.4

Thanks

**Comment by MarcinZiabek** [comment]
Could you please share the SVG file with me? 
Is the image embedded or provided as a path to local storage? 
Could you please validate your SVG file against the official specification? https://validator.w3.org

**Comment by lukasvecerka23** [comment]
Yes, here is the SVG file together with generated PDF.

It seems that some of the embedded pngs are generated but some of them not.

I tried to validate the SVG using provided validator but I'm getting 429 Too Many Requests.

![background2](https://github.com/user-attachments/assets/4df73033-7e79-4c3d-a776-54095f15e701)

[test.pdf](https://github.com/user-attachments/files/16189825/test.pdf)

**Comment by MarcinZiabek** [comment]
When I open your SVG file in Safar or Chrome, specific images are also not visible. It seems that your SVG file may contain issues that only some SVG parsers can handle.

<img width="792" alt="image" src="https://github.com/user-attachments/assets/3c7a80b6-3558-404f-847a-b0c97b69cdf9">

I have opened your SVG in Affinity Designer and resaved it. After this operation, Chrome shows the file correctly. Please try it:

[fixed.svg.zip](https://github.com/user-attachments/files/16191109/fixed.svg.zip)

**Comment by lukasvecerka23** [comment]
Yeah now it works. The original SVG was converted from PDF so perhaps the converter doesn't work correctly. Do you have any idea how could be PDF vectors converted to SVG easily?

Thanks for help!

**Comment by MarcinZiabek** [comment]
It's not easy... PDF does not support embedding SVG files. QuestPDF (through the Skia library) translates SVG contents into PDF drawing commands. So, the opposite process must be equally difficult: analyze every drawing command and find the nearest feature from SVG. It is wholly expected to produce not-perfect results.

**Comment by lukasvecerka23** [comment]
Thanks for explanation. And are you planning to add support to use PDF images as a background in QuestPDF? That would be helpful.

**Comment by MarcinZiabek** [comment]
Could you please elaborate a bit more about that feature?

**Comment by amitjaura** [comment]
Hi, 
I'm also facing same issue. I simply want to add an image as Page Background. Is there a direct way for this?

**Comment by girlpunk** [comment]
This is already possible using layers. For example:

```csharp
page.Content()
    .Layers(layers =>
    {
        // layer below main content
        layers
            .Layer()
            .Image("path/to/image.png");

        layers
            .PrimaryLayer()
            .Text("Your content goes here");
    });
```

**Comment by MarcinZiabek** [comment]
@girlpunk Thank you so much for answering this question.

Since there has been no recent activity, I’m closing this issue. If you experience a similar problem or have a related request, feel free to open a new GitHub issue. I appreciate your feedback! ❤️
